packages/muelu/src/MueCentral/MueLu_VerboseObject.hpp

```diff
@@ -133,9 +133,7 @@ namespace MueLu {
   private:
     //! Verbose level specific to 'this'
     VerbLevel verbLevel_;
-    mutable
-      int procRank_;
-    int numProcs_;
+    static inline int procRank_ = -1;
```

Comment:
Actually I don't think making it static is correct here. Wouldn't we want different objects to have different verbose ranks?

We could change `SetMueLuOFileStream` (the reason why I made this static) to either take rank as an additional input, or use `Teuchos::DefaultComm` to obtain current rank?

What do you think?

Reply:
`SetMueLuOFileStream` is static so we can't use the non-static `procRank_` there. It used to get the rank using `MPI_COMM_WORLD` which I've changed (initially using the `MueLu::GlobalComm` and now to use the verbose rank (which enforces procRank_ to be static aswell)

Reply:
Ah. I think it might actually be good to use filenames that are labeled using the global communicator. We have spots where we reduce the problem to a sub-communicator. So some ranks could be rank k on the global communicator and rank j in the sub-communicator. And we probably want that output to end up in the same file.

Reply:
> Ah. I think it might actually be good to use filenames that are labeled using the global communicator. We have spots where we reduce the problem to a sub-communicator. So some ranks could be rank k on the global communicator and rank j in the sub-communicator. And we probably want that output to end up in the same file.

I agree.  Rank with respect to MPI_COMM_WORLD is the only thing that makes sense to me for post-run analysis.